src/block-management/block-manager.js

```diff
@@ -66,14 +66,20 @@ export default class BlockManager extends React.Component<PropsType, StateType>
 
 	findDataSourceIndexForFocusedItem() {
 		for ( let i = 0; i < this.state.dataSource.size(); ++i ) {
-			const block = this.state.dataSource.get( i );
+			const block = this.findBlockAtIndex( i );
 			if ( block.focused === true ) {
 				return i;
 			}
 		}
 		return -1;
 	}
 
+	findBlockAtIndex( index: number ): Object {
+		return Platform.OS === 'ios' ?
```

Comment:
I think that the root cause of the issue at hand is that the two lists (the one in the props and the other that's kept in the dataSource) get out of sync. And I think they get out of sync [here](https://github.com/wordpress-mobile/gutenberg-mobile/blob/master/src/store/reducers/index.js#L74-L77), where a block is copied to a new instance and replaced in the list. The `dataSource` is still having the old version of the block in that index and then the focusing action fails to "select" both instances.

That syncing problem can be solved if we mutate the block instead of recreating it, by only setting its attrbibutes:

in `/src/store/reducers/index.js`:
```diff
-                       const index = findBlockIndex( blocks, action.clientId );
-                       blocks[ index ] = {
-                               ...block,
-                               attributes: nextAttributes,
-                       };
+                       block.attributes = nextAttributes;
```

Wdyt?

Reply:
This is great! I completely missed that.
I tested it and works perfectly. Pushing a commit with the change.

Thank you!